src/Analysis/Engine/Impl/Infrastructure/Extensions/HashSetExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.PythonTools.Analysis.Infrastructure.Extensions {
+    public static class HashSetExtensions {
+        public static void Add<T>(this HashSet<T> set, IEnumerable<T> items) {
+            foreach(var e in items) {
```

Comment:
Super nitpick, but I think the formatting rules should have added a space after `foreach`, so hopefully VS hasn't undone your settings (it does it to me all the time...).